File: src/main/java/com/qtech/bubbles/effect/AttackBoostEffect.java
package com.qtech.bubbles.effect;

import com.qtech.bubbles.common.AttributeMap;
import com.qtech.bubbles.common.effect.StatusEffect;
import com.qtech.bubbles.common.effect.StatusEffectInstance;
import com.qtech.bubbles.entity.Entity;
import com.qtech.bubbles.entity.attribute.Attribute;
import com.qtech.utilities.python.builtins.ValueError;

public class AttackBoostEffect extends StatusEffect {
    public AttackBoostEffect() throws ValueError {
        super();
    }

    @Override
    public void execute(Entity entity, StatusEffectInstance statusEffectInstance) {

    }

    @Override
    protected void updateStrength() {

    }

    @Override
    protected boolean canExecute(Entity entity, StatusEffectInstance statusEffectInstance) {
        return false;
    }

    @Override
    public AttributeMap getAttributeModifiers() {
        AttributeMap attributes = new AttributeMap();
        attributes.setBase(Attribute.ATTACK, 1f);

        return attributes;
    }
}

File: src/main/kotlin/com/qtech/bubbles/effect/AttackBoostEffect.kt
package com.qtech.bubbles.effect

import com.qtech.bubbles.common.AttributeMap
import com.qtech.bubbles.common.effect.Effect
import com.qtech.bubbles.common.effect.EffectInstance
import com.qtech.bubbles.common.entity.Attribute
import com.qtech.bubbles.entity.Entity

class AttackBoostEffect : Effect() {
    override fun execute(entity: Entity, effectInstance: EffectInstance) {}
    override fun updateStrength() {}
    override fun canExecute(entity: Entity, effectInstance: EffectInstance): Boolean {
        return false
    }

    override val attributeModifiers: AttributeMap
        get() {
            val attributes = AttributeMap()
            attributes.set(Attribute.ATTACK, 1f)
            return attributes
        }
}